### PR TITLE
task_DEV-3

### PR DESCRIPTION
**Pull request by SatsunkevichAlex**

### CODE REVIEW

File: task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    /// This is a class for convert integer numbers to a new scale of notation
+    /// </summary>
+    class ConverterNewNumberSystem
+    {
+        /// <summary>
+        /// This method convert convert integer numbers to a new scale of notation
+        /// and display it
+        /// </summary>
+        /// <param name="NumberToChange">Initial arguments.</param>
+        /// <param name="SystemBase">Initial arguments.</param>
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs]: Bad idea to have the method name identical to class name

File: task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+        public void ConverterToNewNumberSystem(int NumberToChange, int SystemBase)
+        {
+            StringBuilder NewNumber = new StringBuilder();
+            while (NumberToChange > 0)
+            {
+                int remainder = NumberToChange % SystemBase;
+                if (remainder >= 10)
+                {
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs]: What is 55? unclear, use const value

File: task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+                    char letter = (char)(55 + remainder);
+                    NewNumber.Insert(0, letter);
+                }
+                else
+                {
+                    NewNumber.Insert(0, remainder);
+                }
+
+                NumberToChange /= SystemBase;
+            }
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs]: Single responsibility

File: task_DEV-3/task_DEV-3/EntryPoint.cs [hunk 4]
```diff
@@ -0,0 +1,25 @@
+﻿using System;
+
+namespace task_DEV_3
+{
+    class EntryPoint
+    {
+        static void Main(string[] args)
+        {
+            int InputNumber;
+            int BaseOfNewSystem;
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/EntryPoint.cs]: Rework as it was discussed in char, use Try method and handle exceptions

File: task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+using System.Text;
+
+namespace task_DEV_3
+{
+    /// <summary>
+    /// This is a class for convert integer numbers to a new scale of notation
+    /// </summary>
+    class ConverterNewNumberSystem
+    {
+        /// <summary>
```
Comment (IgorKheidorov) [inline task_DEV-3/task_DEV-3/ConverterNewNumberSystem.cs]: grammar

File: task_DEV-3/task_DEV-3/Converter.cs [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    class Converter
+    {
+        /// <summary>
+        /// This method converts integer numbers to a new scale of notation
+        /// </summary>
+        /// <param name="NumberToChange">Initial arguments.</param>
+        /// <param name="SystemBase">Initial arguments.</param>
+        public string ToNewNumberSystem(int NumberToChange, int SystemBase)
+        {
+            const int BeginOFAlphabetASCII = 55;
```
Comment (AlexanderSoroka) [inline task_DEV-3/task_DEV-3/Converter.cs]: wrong variable name

File: task_DEV-3/task_DEV-3/Converter.cs [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    /// This is a class for convert integer numbers to a new scale of notation
+    /// </summary>
+    class Converter
+    {
+        /// <summary>
+        /// This method converts integer numbers to a new scale of notation
+        /// </summary>
+        /// <param name="NumberToChange">Initial arguments.</param>
+        /// <param name="SystemBase">Initial arguments.</param>
+        public string ToNewNumberSystem(int NumberToChange, int SystemBase)
```
Comment (AlexanderSoroka) [inline task_DEV-3/task_DEV-3/Converter.cs]: codestyle violation

File: task_DEV-3/task_DEV-3/EntryPoint.cs [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+            try
+            {
+                InputNumber = Int32.Parse(args[0]);
```
Comment (AlexanderSoroka) [inline task_DEV-3/task_DEV-3/EntryPoint.cs]: invalid exception handling

**Pull request merged**